source/common/tracing/opentracing_driver_impl.cc

```diff
@@ -128,7 +128,13 @@ OpenTracingDriver::OpenTracingDriver(Stats::Store& stats)
     : tracer_stats_{OPENTRACING_TRACER_STATS(POOL_COUNTER_PREFIX(stats, "tracing.opentracing."))} {}
 
 SpanPtr OpenTracingDriver::startSpan(const Config&, Http::HeaderMap& request_headers,
-                                     const std::string& operation_name, SystemTime start_time) {
+                                     const std::string& operation_name, SystemTime start_time,
+                                     const Tracing::Decision& tracing_decision) {
+  // If tracing decision is no, and sampling decision is not communicated via tags, then
+  // return a null span to indicate that tracing is not being performed.
+  if (!tracing_decision.is_tracing && !useTagForSamplingDecision()) {
+    return nullptr;
```

Comment:
I guess this is related to my other comment. Should we return NullSpan here? I'm not sure if there are other points in the code that might be assuming span is never nullptr? Can you check?

Reply:
This is just replicating the current nullptr span if `is_tracing` is false. But this code will be removed once lightstep supports the sampling tag.

Reply:
Yes, sorry for the delay. I put in https://github.com/lightstep/lightstep-tracer-cpp/pull/94, but there were some concerns around how to handle the propagation of spans with no sampling for lightstep.

I'll follow up with a separate PR once it gets sorted out.